src/bun.js/bindings/KeyObject.cpp

```diff
@@ -0,0 +1,2385 @@
+                    return JSValue::encode(JSC::jsUndefined());
+                }
+                CryptoKeyEC::NamedCurve curve;
+                if (curve_name == NID_X9_62_prime256v1)
+                    curve = CryptoKeyEC::NamedCurve::P256;
+                else if (curve_name == NID_secp384r1)
+                    curve = CryptoKeyEC::NamedCurve::P384;
+                else if (curve_name == NID_secp521r1)
+                    curve = CryptoKeyEC::NamedCurve::P521;
+                else {
+                    EC_KEY_free(ec_key);
+                    throwException(globalObject, scope, createTypeError(globalObject, "Unsupported EC curve"_s));
+                    return JSValue::encode(JSC::jsUndefined());
+                }
+                auto alg = CryptoAlgorithmIdentifier::ECDH;
+                auto result = CryptoKeyEC::platformImportSpki(alg, curve, Vector<uint8_t>((uint8_t*)data, byteLength), true, CryptoKeyUsageVerify);
+                if (result == nullptr) {
+                    alg = CryptoAlgorithmIdentifier::ECDSA;
+                    result = CryptoKeyEC::platformImportSpki(CryptoAlgorithmIdentifier::ECDSA, curve, Vector<uint8_t>((uint8_t*)data, byteLength), true, CryptoKeyUsageVerify);
+                }
+                if (result == nullptr) {
+                    throwException(globalObject, scope, createTypeError(globalObject, "Invalid EC public key"_s));
+                    return JSValue::encode(JSC::jsUndefined());
+                }
+                auto impl = result.releaseNonNull();
+                return JSC::JSValue::encode(JSCryptoKey::create(structure, zigGlobalObject, WTFMove(impl)));
+            } else {
+                throwException(globalObject, scope, createTypeError(globalObject, "Unsupported public key"_s));
+                return JSValue::encode(JSC::jsUndefined());
+            }
+        }
+
+        JSC::throwTypeError(globalObject, scope, "type should be 'pkcs1' or 'spki'"_s);
+        return JSValue::encode(JSC::jsUndefined());
+    }
+    JSC::throwTypeError(globalObject, scope, "format should be 'pem' or 'der'"_s);
+    return JSValue::encode(JSC::jsUndefined());
+}
+
+JSC::EncodedJSValue KeyObject__createSecretKey(JSC::JSGlobalObject* lexicalGlobalObject, JSC::CallFrame* callFrame)
+{
+
+    JSValue bufferArg = callFrame->uncheckedArgument(0);
+    auto& vm = lexicalGlobalObject->vm();
+    auto scope = DECLARE_THROW_SCOPE(vm);
+    Zig::GlobalObject* globalObject = reinterpret_cast<Zig::GlobalObject*>(lexicalGlobalObject);
+    auto* structure = globalObject->JSCryptoKeyStructure();
+
+    if (!bufferArg.isCell()) {
+        throwException(lexicalGlobalObject, scope, createTypeError(lexicalGlobalObject, "ERR_INVALID_ARG_TYPE: expected Buffer or array-like object"_s));
+        return JSValue::encode(JSC::jsUndefined());
+    }
+
+    auto type = bufferArg.asCell()->type();
+
+    switch (type) {
+
+    case DataViewType:
```

Comment:
at some point we should add a macro for this

Float16Array will be a thing in like a year probably

https://github.com/tc39/proposal-float16array